test/e2e/metamask.spec.js

```diff
@@ -243,93 +243,94 @@ describe('Metamask popup page', function () {
-      const byMetamaskSubmit = By.css('#pending-tx-form > div.flex-row.flex-space-around.conf-buttons > input')
-      const metamaskSubmit = await driver.wait(until.elementLocated(byMetamaskSubmit))
-      await metamaskSubmit.click()
-      await delay(1000)
-    })
-
-    it('switches back to Token Factory to grab the token contract address', async function () {
-      const windowHandles = await driver.getAllWindowHandles()
-      await driver.switchTo().window(windowHandles[0])
-      const tokenContactAddress = await driver.findElement(By.css('#main > div > div > div > div:nth-child(2) > span:nth-child(3)'))
-      tokenAddress = await tokenContactAddress.getText()
-      await delay(500)
-    })
-
-    it('navigates back to MetaMask popup in the tab', async function () {
-      if (process.env.SELENIUM_BROWSER === 'chrome') {
-        await driver.get(`chrome-extension://${extensionId}/popup.html`)
-      } else if (process.env.SELENIUM_BROWSER === 'firefox') {
-        await driver.get(`moz-extension://${extensionId}/popup.html`)
-      }
-      await delay(700)
-    })
-  })
-
-  describe('Add Token', function () {
-
-    it('switches to the add token screen', async function () {
-      const tokensTab = await driver.findElement(By.css('#app-content > div > div.app-primary.from-right > div > section > div > div.inactiveForm.pointer'))
-      assert.equal(await tokensTab.getText(), 'TOKENS')
-      await tokensTab.click()
-      await delay(300)
-    })
-
-    it('navigates to the add token screen', async function () {
-      const addTokenButton = await driver.findElement(By.css('#app-content > div > div.app-primary.from-right > div > section > div.full-flex-height > div > button'))
-      assert.equal(await addTokenButton.getText(), 'ADD TOKEN')
-      await addTokenButton.click()
-    })
-
-    it('checks add token screen rendered', async function () {
-      const addTokenScreen = await driver.findElement(By.css('#app-content > div > div.app-primary.from-right > div > div.section-title.flex-row.flex-center > h2'))
-      assert.equal(await addTokenScreen.getText(), 'ADD TOKEN')
-    })
-
-    it('adds token parameters', async function () {
-      const tokenContractAddress = await driver.findElement(By.css('#token-address'))
-      await tokenContractAddress.sendKeys(tokenAddress)
-      await delay(300)
-      await driver.findElement(By.css('#app-content > div > div.app-primary.from-right > div > div.flex-column.flex-justify-center.flex-grow.select-none > div > button')).click()
-      await delay(200)
-    })
-
-    it('checks the token balance', async function () {
-      const tokenBalance = await driver.findElement(By.css('#app-content > div > div.app-primary.from-left > div > section > div.full-flex-height > ol > li:nth-child(2) > h3'))
-      assert.equal(await tokenBalance.getText(), '100 TST')
-    })
-  })
+  // describe('Token Factory', function () {
```

Comment:
Would prefer to include the injection request as part of the e2e tests. Maybe @tmashuang could take a pass at this?